### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by cherryfield**

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "main" ]
+  schedule:
+    - cron: '42 21 * * 1'
```
Comment (davidogbiko) [inline .github/workflows/codeql-analysis.yml:21]: 9:42pm... 🤔 , looks strange. Any particular reason we choose this time?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        
+        # Details on CodeQL's query packs refer to : https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#using-queries-in-ql-packs
+        # queries: security-extended,security-and-quality
+
+        
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
```
Comment (davidogbiko) [inline .github/workflows/codeql-analysis.yml:58]: I think we might not need this step for now.

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (cherryfield) [reply .github/workflows/codeql-analysis.yml:21]: No particular reason.

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (cherryfield) [reply .github/workflows/codeql-analysis.yml:58]: Okay

**Pull request merged**